cuda_core/cuda/core/experimental/_linker.py

```diff
@@ -395,32 +395,56 @@ def __init__(self, *object_codes: ObjectCode, options: LinkerOptions = None):
 
     def _add_code_object(self, object_code: ObjectCode):
         data = object_code._module
-        assert_type(data, bytes)
         with _exception_manager(self):
             name_str = f"{object_code.name}"
-            if _nvjitlink:
-                _nvjitlink.add_data(
-                    self._mnff.handle,
-                    self._input_type_from_code_type(object_code._code_type),
-                    data,
-                    len(data),
-                    name_str,
-                )
-            else:
-                name_bytes = name_str.encode()
-                handle_return(
-                    _driver.cuLinkAddData(
+            if isinstance(data, str):
+                # Handle file path input
+                if _nvjitlink:
```

Comment:
For the nested if here, we now have 4 blocks. I'd like to flatten them, starting with
```
if _nvjitlink and isinstance(bytes):
    ...
elif _nvjitlink and isinstance(str):
    ...
elif <omitted>:
    ...
else:
    # keep raising the type error
```